server/chat-plugins/help.ts

```diff
@@ -0,0 +1,467 @@
+			if (!this.can('ban', null, helpRoom)) return false;
+			// intended for use mainly within the page, so supports being used in all rooms
+			this.room = helpRoom;
+			const index = parseInt(target) - 1;
+			if (isNaN(index)) return this.errorReply(`Invalid queue index.`);
+			const str = Answerer.queue[index];
+			if (!str) return this.errorReply(`Item does not exist in queue.`);
+			const regex = Answerer.stringRegex(str);
+			// validated on submission
+			const faq = Answerer.getFaqID(str.split('=>')[1].trim());
+			if (!faq) return this.errorReply(`Invalid FAQ.`);
+			if (!Answerer.data.pairs[faq]) helpData.pairs[faq] = [];
+			Answerer.data.pairs[faq].push(regex);
+			Answerer.queue.splice(index, 1);
+			Answerer.writeState();
+
+			this.privateModAction(`${user.name} approved regex for use with queue number ${target}`);
+			this.modlog(`HELPFILTER APPROVE`, null, `${target}`);
+		},
+		deny(target, room, user) {
+			if (!helpRoom) return this.errorReply(`No Help room to manage the filter for.`);
+			if (!this.can('ban', null, helpRoom)) return false;
+			// intended for use mainly within the page, so supports being used in all rooms
+			this.room = helpRoom;
+			target = target.trim();
+			const index = parseInt(target) - 1;
+			if (isNaN(index)) return this.errorReply(`Invalid queue index.`);
+			if (!Answerer.queue[index]) throw new Chat.ErrorMessage(`Item does not exist in queue.`);
+			Answerer.queue.splice(index, 1);
+			Answerer.writeState();
+			this.privateModAction(`${user.name} denied regex with queue number ${target}`);
+			this.modlog(`HELPFILTER DENY`, null, `${target}`);
+		},
+	},
+	helpfilterhelp() {
+		const help = [
+			`<code>/helpfilter stats</code> - Shows stats for the Help filter (matched lines and the FAQs that match them.)`,
+			`<code>/helpfilter keys</code> - View regex keys for the Help filter.`,
+			`<code>/helpfilter toggle [on | off]</code> - Enables or disables the Help filter. Requires: @ # &`,
+			`<code>/helpfilter add [input] => [faq]</code> - Adds regex made from the input string to the Help filter, to respond with [faq] to matches.`,
+			`<code>/helpfilter remove [faq], [regex index]</code> - removes the regex matching the [index] from the Help filter's responses for [faq].`,
+			`<code>/helpfilter queue [regex] => [faq]</code> - Adds [regex] for [faq] to the queue for Help staff to review.`,
+			`<code>/helpfilter approve [index]</code> - Approves the regex at position [index] in the queue for use in the Help filter.`,
+			`<code>/helpfilter deny [index]</code> - Denies the regex at position [index] in the Help filter queue.`,
+			`Indexes can be found in /helpfilter keys.`,
+			`Requires: @ # &`,
+		];
+		return this.sendReplyBox(help.join('<br/ >'));
+	},
+};
+
+export const pages: PageTable = {
+	helpfilter(args, user) {
+		if (!helpRoom) return `<h2>No Help room to view data for.</h2>`;
+		const canChange = user.authAtLeast(`@`, helpRoom);
```

Comment:
```suggestion
		const canChange = helpRoom.auth.atLeast(user, `@`);
```
User#authAtLeast is being removed on next restart in https://github.com/smogon/pokemon-showdown/pull/7046.